### ISSUE TITLE
Crash after playing sounds

### ISSUE THREAD

**Issue by dougo** [angstsmurf/spatterlight#77]
I'm playtesting a game for someone, and Spatterlight crashed after playing some sounds. Forum thread here: https://intfiction.org/t/diagnosing-a-possibly-sound-related-or-timer-related-crash-in-spatterlight-glulx/69879/6

I haven't yet been able to reproduce it, but here's the crash log in case it helps:
[Spatterlight-2024-07-23-002849.ips.json](https://github.com/user-attachments/files/16354346/Spatterlight-2024-07-23-002849.ips.json)

**Comment by erkyrath** [comment]
The relevant part of this seems to be:

```
Thread 7 Crashed:: org.sbooth.AudioEngine.Decoder
0   SFBAudioEngine                	       0x103267a08 SFB::Audio::RingBuffer::WriteAudio(AudioBufferList const*, unsigned long) + 216
1   SFBAudioEngine                	       0x103272b40 SFB::Audio::Player::DecoderThreadEntry() + 2528
2   SFBAudioEngine                	       0x103274cb4 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct>>, void* (SFB::Audio::Player::*)(), SFB::Audio::Player*>>(void*) + 64
3   libsystem_pthread.dylib       	       0x18495ef94 _pthread_start + 136
4   libsystem_pthread.dylib       	       0x184959d34 thread_start + 8
```

**Comment by angstsmurf** [comment]
Thanks for reporting this.

The crash seems to have occurred in a background thread created by the SFBAudioEngine library, which I am not very familiar with. Without a way to reproduce it, this crash is going to be difficult to debug.

**Comment by erkyrath** [comment]
I don't know anything about the library either. The obvious blind guess is "something got deallocated while the sound-playing machinery was in the middle of using it".

**Comment by angstsmurf** [comment]
Yes, I've had reproducible crashes in the past when a game issued a sound or a volume change with a notification callback, and then the interpreter process was killed right before the notification happened.

This was the main reason that I switched from SDL audio to SFBAudioEngine, in the hope that the automatic reference counting would help, and I never saw any audio related crashes after that. Until now.

**Comment by dougo** [comment]
> EDIT 3: By the way, @erkyrath, what do you use to parse the `.json` file? EDIT 4: Oh, right, I just had to remove the `.json` suffix to make Console and Xcode parse it correctly.

Sorry, forgot to mention that I had to add the .json suffix to get Github to accept the upload.

**Comment by dougo** [comment]
Sorry, still no luck reproducing it. 🤷‍♂️
